src/core/control/Control.cpp

```diff
@@ -1352,11 +1348,11 @@ void Control::updateDeletePageButton() {
 void Control::deletePage() {
     clearSelectionEndText();
 
-    // if the current page contains the Setsquare, reset it
+    // if the current page contains the geometry tool, reset it
     size_t pNr = getCurrentPageNo();
-    auto setsquareView = win->getXournal()->getSetsquareView();
-    if (setsquareView && doc->indexOf(setsquareView->getPage()) == pNr) {
-        win->getXournal()->resetSetsquareView();
+    auto geometryToolController = win->getXournal()->getGeometryToolController();
+    if (geometryToolController && doc->indexOf(geometryToolController->getPage()) == pNr) {
```

Comment:
Not sure, but we may want the lock the document mutex for that

Reply:
Fine

Reply:
Done